pkg/user/signer.go

```diff
@@ -153,7 +153,7 @@ func (s *Signer) SubmitPayForBlob(ctx context.Context, blobs []*blob.Blob, opts
 		return nil, err
 	}
 	if resp.Code != 0 {
-		return resp, fmt.Errorf("tx failed with code %d: %s", resp.Code, resp.RawLog)
+		return nil, fmt.Errorf("tx failed with code %d: %s", resp.Code, resp.RawLog)
```

Comment:
blocking

wait this change seems unrelated. why do we want to return nil and not the response here?

Reply:
Thanks @evan-forbes for your review!
I see here we are already returning an error with a code, so there is no need to return a response. Returning nil would be better.

Reply:
And I don't understand why the old code returns response here.

Reply:
your point is definitley valid, where ideally we would return only the tx response or nil and an error. Since the sdk includes so much information in the tx response, I think we need to keep returning it as well. We can't get rid of the error entirely either since there are instances where the sdk will return a nil tx response. Perhaps we can not perform the check on the code to keep things logically cleaner, however that check is performed in almost all cases.

regardless, that change is outside of the scope of the refactor imo, and changing the return value seems very close to a breaking change if it isn't one. Existing scripts that I've written would at least result in a panic should this get hit.

Reply:
we can see this actually break the `TestLegacyUpgrade` test

Reply:
I'd approve without this change, and we should open an issue to track the returning nil thing

Reply:
Ok, thanks! I will ignore this change.

Reply:
Done @evan-forbes